commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/xml/deserializer/stream/FileListXmlStreamDeserializer.kt

```diff
@@ -0,0 +1,35 @@
+package ac.uk.ebi.biostd.xml.deserializer.stream
+
+import com.fasterxml.jackson.dataformat.xml.JacksonXmlModule
+import com.fasterxml.jackson.dataformat.xml.XmlMapper
+import ebi.ac.uk.model.Attribute
+import ebi.ac.uk.model.File as PageTabFile
+import ebi.ac.uk.model.FileList
+import java.io.File
+import javax.xml.stream.XMLInputFactory
+
+internal class FileListXmlStreamDeserializer {
+    fun deserialize(file: File): FileList {
+        val referencedFiles: MutableList<PageTabFile> = mutableListOf()
+        val reader = XMLInputFactory.newFactory().createXMLStreamReader(file.inputStream())
+        val module = JacksonXmlModule().apply {
+            setDefaultUseWrapper(false)
+            addDeserializer(PageTabFile::class.java, FileXmlStreamDeserializer())
+            addDeserializer(Attribute::class.java, AttributeXmlStreamDeserializer())
+        }
+        val mapper = XmlMapper(module)
+        reader.next()
+
+        while (reader.hasNext()) {
```

Comment:
here and in others part of the code this "old" style iterations make the code looks pretty complicate and obfuscate logic. is true that the library provides and horrible api but we can extend it to provide a nice method. For example, here I can imagine.

`Reader.forEach(element: (ReaderType) -> Unit)`